packages/block-editor/src/components/block-list/use-block-props/use-block-refs.js

```diff
@@ -18,6 +18,50 @@ import { BlockRefs } from '../../provider/block-refs-provider';
 /** @typedef {import('@wordpress/element').RefCallback} RefCallback */
 /** @typedef {import('@wordpress/element').RefObject} RefObject */
 
+function addToMap( map, id, value ) {
+	let setForId = map.get( id );
+	if ( ! setForId ) {
+		setForId = new Set();
+		map.set( id, setForId );
+	}
+	setForId.add( value );
+}
+
+function deleteFromMap( map, id, value ) {
+	const setForId = map.get( id );
+	if ( ! setForId ) {
+		return;
+	}
+	setForId.delete( value );
+	if ( setForId.size === 0 ) {
+		map.delete( id );
+	}
+}
+
+function getRefElement( refs, clientId ) {
+	// Multiple refs may be created for a single block. Find the
+	// first that has an element set.
```

Comment:
I don't remember why there were multiple refs 😬

Reply:
There are multiple refs if you call `useBlockRefProvider( clientId )` twice for the same `clientId`:
```js
const refA = useBlockRefProvider( clientId );
return <div ref={ refA } />;

// somewhere else, with same clientId but different element
const refB = useBlockRefProvider( clientId );
return <div ref={ refB } />;
```
The it registers two refs for the same `clientId`, and `useBlockRef`/`useBlockElement` will return the one that's non-null.

I also don't know if this ever happens 🙂

Reply:
I wonder if we should just error if we try to call it twice with the same clientId

Reply:
Yes, ignoring the second call and printing a warning would certainly simplify the code. Then we can store just one element instead of a `Set`.